test/threads.jl

```diff
@@ -322,6 +322,19 @@ close(proc.in)
     end
 end
 
+@testset "threads clean up task local state" begin
```

Comment:
Note that in my tests in WorkerUtilities.jl, I couldn't get the `task.code= nothing` solution to work when the tests were run inside a `@testset`; I did a little digging, but couldn't figure out what was going on there. Possibly something related to how `@testset`s use task-local-storage maybe?

Reply:
🤔 weirdly the tests passed here though, it seems?